libcodechecker/log/option_parser.py

```diff
@@ -308,218 +200,228 @@ def item(self):
-        set_attr('^-(E|M[T|Q|F|J|P|V|M]*)$', result, 'action',
-                 ActionType.PREPROCESS, True),
-        set_attr('-print-prog-name', result, 'action', ActionType.INFO),
-        append_to_list(COMPILE_OPTION_MAP, result.compile_opts),
-        append_to_list(COMPILER_LINKER_OPTION_MAP, result.link_opts),
-        append_to_list(LINKER_OPTION_MAP_REGEX, result.link_opts, True),
-        append_to_list(COMPILE_OPTION_MAP_REGEX, result.compile_opts, True),
-        append_merged_to_list(COMPILE_OPTION_MAP_MERGED, result.compile_opts),
-        append_merged_to_list(LINK_OPTION_MAP_MERGED, result.link_opts),
-        skip(LINKER_OPTION_MAP),
-        append_to_list_from_file('-filelist', result.files),
-        append_to_list({'^[^-].+': 0}, result.files, True)]
-
-    for coll in arg_collection:
-        res = coll(it.item)
-        if res:
-            return True
+def get_language(extension):
+    # TODO: There are even more in the man page of gcc.
+    mapping = {'.c': 'c',
+               '.cp': 'c++',
+               '.cpp': 'c++',
+               '.cxx': 'c++',
+               '.txx': 'c++',
+               '.cc': 'c++',
+               '.C': 'c++',
+               '.ii': 'c++',
+               '.m': 'objective-c',
+               '.mm': 'objective-c++'}
+    return mapping.get(extension)
+
+
+def __append_from_file(flag_iterator, target, buildaction):
+    if flag_iterator.item == '-filelist':
+        next(flag_iterator)
+
+        with open(flag_iterator.item) as f:
+            for line in f:
+                # TODO: What kind of flags are these? Shouldn't these be
+                # appended to the compilation options collected in the
+                # buildaction object?
+                target.append(line.strip())
+
+        return True
 
-    # Unhandled compilation argument found.
-    LOG.debug("Unhandled argument: " + str(it.item))
     return False
 
 
-def parse_options(args):
-    """ Requires a full compile command with the compiler,
-    not only arguments."""
+def __collect_compile_opts(flag_iterator, target, buildaction):
+    """
+    This function collects the compilation (i.e. not linker or preprocessor)
+    flags to the buildaction and saves to the target command.
+    """
+    if COMPILE_OPTIONS.match(flag_iterator.item):
+        target.append(flag_iterator.item)
+        buildaction.analyzer_options.append(flag_iterator.item)
+        return True
 
-    # Keep " characters.
-    args = args.replace(r'"', r'"\"')
+    m = COMPILE_OPTIONS_MERGED.match(flag_iterator.item)
 
-    result_map = OptionParserResult()
+    if m:
+        target.append(flag_iterator.item)
+        buildaction.analyzer_options.append(flag_iterator.item)
 
-    # The first element in the list is the compiler skip it from parsing.
-    for it in OptionIterator(shlex.split(args)[1:]):
-        arg_check(it, result_map)
+        if len(m.group(0)) == len(flag_iterator.item):
```

Comment:
Could you add a comment what is this check actually doing?

Reply:
This commented code fragment is outdated by now. In the new version this is more intuitive.